src/unittest/test_collision.cpp

```diff
@@ -132,48 +132,48 @@ void TestCollision::testAxisAlignedCollision()
 			aabb3f s(bx, by, bz, bx+2, by+2, bz+2);
 			aabb3f m(bx+2.3, by+2.29, bz+2.29, bx+4.2, by+4.2, bz+4.2);
 			v3f v(-1./3, -1./3, -1./3);
-			f32 dtime;
-			UASSERT(axisAlignedCollision(s, m, v, 0, &dtime) == 0);
+			f32 dtime = 1.0f;
+			UASSERT(axisAlignedCollision(s, m, v, &dtime) == 0);
 			UASSERT(fabs(dtime - 0.9) < 0.001);
 		}
 		{
 			aabb3f s(bx, by, bz, bx+2, by+2, bz+2);
 			aabb3f m(bx+2.29, by+2.3, bz+2.29, bx+4.2, by+4.2, bz+4.2);
 			v3f v(-1./3, -1./3, -1./3);
-			f32 dtime;
-			UASSERT(axisAlignedCollision(s, m, v, 0, &dtime) == 1);
+			f32 dtime = 1.0f;
+			UASSERT(axisAlignedCollision(s, m, v, &dtime) == 1);
 			UASSERT(fabs(dtime - 0.9) < 0.001);
 		}
 		{
 			aabb3f s(bx, by, bz, bx+2, by+2, bz+2);
 			aabb3f m(bx+2.29, by+2.29, bz+2.3, bx+4.2, by+4.2, bz+4.2);
 			v3f v(-1./3, -1./3, -1./3);
-			f32 dtime;
-			UASSERT(axisAlignedCollision(s, m, v, 0, &dtime) == 2);
+			f32 dtime = 1.0f;
+			UASSERT(axisAlignedCollision(s, m, v, &dtime) == 2);
 			UASSERT(fabs(dtime - 0.9) < 0.001);
 		}
 		{
 			aabb3f s(bx, by, bz, bx+2, by+2, bz+2);
 			aabb3f m(bx-4.2, by-4.2, bz-4.2, bx-2.3, by-2.29, bz-2.29);
 			v3f v(1./7, 1./7, 1./7);
-			f32 dtime;
-			UASSERT(axisAlignedCollision(s, m, v, 0, &dtime) == 0);
+			f32 dtime = 17.0f;
+			UASSERT(axisAlignedCollision(s, m, v, &dtime) == 0);
 			UASSERT(fabs(dtime - 16.1) < 0.001);
 		}
 		{
 			aabb3f s(bx, by, bz, bx+2, by+2, bz+2);
 			aabb3f m(bx-4.2, by-4.2, bz-4.2, bx-2.29, by-2.3, bz-2.29);
 			v3f v(1./7, 1./7, 1./7);
-			f32 dtime;
-			UASSERT(axisAlignedCollision(s, m, v, 0, &dtime) == 1);
+			f32 dtime = 17.0f;
```

Comment:
where do these values come from?

Reply:
dtime parameter has been changed to input/output.
On input it signifies the time left in the step, here it is set as needed for the collision to occur according to speed and distance.

This is a performance optimization, previously calculations were run on all boxes in radius (box extent +1.5n), now boxes are rejected based on distance between faces > dtime*speed